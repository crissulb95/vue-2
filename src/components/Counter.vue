<template>
  <div class="counter-container">
    <h3 class="counter-title">{{ customTitle }}</h3>
    <p>{{ counter }}<sup>2</sup> = {{ getSquareCounter }}</p>
    <p data-testId="counter">{{ counter }}</p>
    <div class="counter-btn-container">
      <button @click="increase">+1</button>
      <button @click="decrease">-1</button>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: 'Title'
    },
    start: {
      type: Number,
      default: 5,
      validator(value) {
        return value >= 0
      }
    }
  },
  /* props:[title] //Si no quieres especificar el tipado */
  name: 'Counter',
  data() {
    return {
      counter: this.start
    }
  },
  methods: {
    //Rerenderea cada vez que se invoca
    /* squareCounter() {
      console.log("from methods");
      return this.counter * this.counter;
    }, */
    increase() {
      return this.counter++
    },
    decrease() {
      return this.counter--
    }
  },
  computed: {
    //Renderea y cachea el resultado; cambia solo cuando ocurre
    //alg√∫n cambio en las dependencias
    getSquareCounter() {
      //console.log("from computed");
      return this.counter * this.counter
    },
    customTitle() {
      return this.title || 'Default title'
    }
  }
}
</script>

<style scoped>
.counter-container {
  display: block;
  width: 50%;
  flex: 0 0 50%;
  padding: 50px;
}
.counter-title {
  margin-top: 50px;
  font-size: 3rem;
}
.counter-btn-container {
  display: flex;
  width: 100%;
  justify-content: center;
  margin-top: 50px;
}
.counter-btn-container > button {
  background-color: transparent;
  border-radius: 5px;
  border: 1px solid white;
  color: white;
  cursor: pointer;
  font-size: 1.3rem;
  margin-right: 20px;
  padding: 5px 30px;
  transition: 0.4s all ease-in-out;
}
.counter-btn-container > button:last-child {
  margin-right: 0;
}
.counter-btn-container > button:hover {
  background-color: lightgreen;
  border-color: lightgreen;
  color: darkslategray;
}
</style>
