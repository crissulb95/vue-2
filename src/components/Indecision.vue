<template>
  <div class="indecision-container">
    <figure class="img-container">
      <img v-if="bgGif" :src="bgGif" alt="bg" />
    </figure>
    <div class="input-container">
      <input
        type="text"
        placeholder="Haz una pregunta"
        v-model="question"
        data-test-id="input-question"
      />
      <small>Recuerda terminar con un signo de interrogaci√≥n (?)</small>
    </div>
    <div v-if="isValidQuestion" class="question-container">
      <p>{{ question }}</p>
      <h3>{{ answer }}</h3>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      question: "primer valor",
      answer: null,
      bgGif: null,
      isValidQuestion: false,
    };
  },
  methods: {
    async getAnswer() {
      try {
        this.answer = "pensando...";

        const { answer, image } = await fetch("https://yesno.wtf/api")
          .then((res) => res.json())
          .catch((err) => console.error(err));

        this.answer =
          answer === "yes" ? "Si" : answer === "maybe" ? "Tal vez" : "No!";
        this.bgGif = image;
      } catch (error) {
        this.answer = "Hubo un error en la carga desde el API";
        this.bgGif = null;
        console.log(
          "IndecisionComponent: Error en la carga desde el API",
          error
        );
      }
    },
  },
  watch: {
    question(val) {
      this.isValidQuestion = false;
      console.log({ val });

      if (!val.includes("?")) return false;

      console.log({ val });
      this.isValidQuestion = true;
      this.getAnswer();
    },
  },
};
</script>

<style scoped>
.indecision-container {
  display: block;
  width: 50%;
  flex: 0 0 50%;
  position: relative;
  padding: 150px 20px;
  margin: 0;
}

.indecision-container .img-container {
  height: 100%;
  left: 50%;
  margin: 0;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  z-index: 0;
}
.indecision-container .img-container::after {
  background-color: rgba(0, 0, 0, 0.3);
  content: "";
  height: 100%;
  left: 50%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  z-index: 1;
}
.indecision-container .img-container img {
  height: 100%;
  width: 100%;
  position: absolute;
  top: 50%;
  left: 50%;
  object-fit: cover;
  object-position: center;
  transform: translate(-50%, -50%);
}

.input-container {
  position: relative;
  z-index: 2;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
}

.input-container > input {
  padding: 15px;
  flex: 0 0 50%;
  border-radius: 5px;
  margin-bottom: 10px;
}
.input-container > small {
  flex: 0 0 100%;
}

.question-container {
  display: flex;
  flex: 0 0 100%;
  position: relative;
  z-index: 2;
  flex-wrap: wrap;
  margin-top: 30px;
}
.question-container p,
.question-container h3 {
  display: block;
  width: 100%;
  flex: 0 0 100%;
  min-height: 20px;
}
</style>
