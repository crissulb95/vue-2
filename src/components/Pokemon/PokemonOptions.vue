<template>
  <div class="pokemon-options-container">
    <div class="message">
      <span v-if="show">
        Es <strong>{{ correctPokemon.name }} !</strong>
      </span>
    </div>
    <ul>
      <li
        v-for="(pokemon, index) in pokemons"
        :key="pokemon.id"
        @click="$emit('selection', pokemon.id)"
        :class="afterClass(pokemon.id)"
        :id="'option-' + (index + 1)"
      >
        <span>{{ pokemon.name }}</span>
      </li>
    </ul>
    <button @click="$emit('newGame')">Volver a jugar !</button>
  </div>
</template>

<script>
export default {
  name: "Pokemon Options",
  props: {
    pokemons: {
      type: Array,
      required: true,
    },
    correctPokemon: {
      type: Object,
      required: true,
    },
    show: {
      type: Boolean,
      required: true,
      default: false,
    },
  },
  methods: {
    afterClass(value) {
      let auxClass = "fade-in";
      if (this.show && value === this.correctPokemon.id)
        return `${auxClass} correct`;
      if (this.show) return `${auxClass} wrong`;
      return auxClass;
    },
  },
};
</script>

<style scoped>
.pokemon-options-container {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}
.pokemon-options-container > .message {
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
  margin-bottom: 0;
  height: 2rem;
}
.pokemon-options-container > .message > span {
  font-size: 1.5rem;
}
.pokemon-options-container > .message > span > strong {
  margin-left: 4px;
  color: lightgreen;
  text-transform: uppercase;
}
.pokemon-options-container ul {
  list-style-type: none;
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  max-width: 1140px;
  justify-content: space-between;
  padding-left: 0;
  position: relative;
}

.pokemon-options-container ul::after {
  position: absolute;
  content: "";
  height: 15px;
  width: 15px;
  background-color: lightgrey;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%) rotate(45deg);
  border-radius: 2px;
  box-shadow: inset 1px 1px 4px 0 white;
  transition: 0.4s all ease-in-out;
}

@media (max-width: 575px) {
  .pokemon-options-container ul {
    order: 2;
  }
  .pokemon-options-container ul::after {
    bottom: 0px;
    top: unset;
    transform: translate(-50%, 50%) rotate(45deg);
  }
}
.pokemon-options-container li {
  align-items: center;
  background-color: white;
  border-radius: 10px;
  color: black;
  cursor: pointer;
  display: flex;
  flex: 0 0 100%;
  flex: 0 0 calc(50% - 10px);
  justify-content: center;
  margin-bottom: 20px;
  padding: 20px 80px;
  position: relative;
  transition: 0.4s all linear;
  width: 100%;
}
.pokemon-options-container li:nth-child(3),
.pokemon-options-container li:nth-child(4) {
  margin-bottom: 0;
}

@media (max-width: 575px) {
  .pokemon-options-container li {
    flex: 0 0 100%;
    width: 100%;
    padding: 20px;
    margin-bottom: 20px !important;
  }
}
.pokemon-options-container li::after {
  background-color: red;
  border-bottom: 4px solid black;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  content: "";
  height: 50%;
  left: 50%;
  opacity: 1;
  position: absolute;
  top: 0%;
  transform: translate(-50%, 0%);
  transition: 0.4s all ease-in-out;
  width: 100%;
  z-index: 0;
}

.pokemon-options-container li > span {
  position: relative;
  display: block;
  z-index: 3;
  padding: 5px;
  flex: 0 0 100%;
  transition: 0.4s all ease-in-out;
  text-transform: capitalize;
  line-height: 1.5rem;
  font-size: 1rem;
}
.pokemon-options-container li > span::after {
  border-radius: 10px;
  border: 4px solid transparent;
  content: "";
  display: block;
  height: 100%;
  left: 50%;
  padding: 10px;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  transition: 0.4s all ease-in-out;
  width: 100%;
  z-index: -1;
  background-color: white;
  border-color: black;
}

.pokemon-options-container li:hover {
  box-shadow: 1px 1px 5px 1px rgba(0, 0, 0, 0.5);
}

.pokemon-options-container li:hover::after {
  height: 0%;
  opacity: 0;
}
.pokemon-options-container li:hover > span::after {
  background-color: transparent;
  border-color: transparent;
}
.pokemon-options-container li:hover > span {
  font-size: 1.3rem;
}
.pokemon-options-container li.wrong {
  background-color: lightpink;
}

.pokemon-options-container li.correct {
  background-color: lightgreen;
}
.pokemon-options-container li.correct::after,
.pokemon-options-container li.wrong::after {
  height: 0;
  opacity: 0;
}
.pokemon-options-container li.correct > span,
.pokemon-options-container li.wrong > span {
  font-size: 1.3rem;
  color: white;
}
.pokemon-options-container li.correct > span::after,
.pokemon-options-container li.wrong > span::after {
  background-color: transparent;
  border-color: transparent;
}

.pokemon-options-container > button {
  position: absolute;
  top: 20px;
  right: 20px;
  border: 2px solid transparent;
  padding: 10px 30px;
  transition: 0.4s all ease-in-out;
  border-radius: 5px;
  background-color: white;
  cursor: pointer;
}
.pokemon-options-container > button:hover {
  background-color: transparent;
  border-color: white;
  color: white;
}

@media (max-width: 575px) {
  .pokemon-options-container > button {
    position: relative;
    right: unset;
    top: unset;
    margin: 20px 0;
    order: 0;
  }
}
</style>
