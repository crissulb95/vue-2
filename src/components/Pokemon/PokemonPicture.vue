<template>
  <figure class="pokemon-picture-container">
    <img :src="imgSrc" alt="pokemon image" class="hidden-pokemon fade-in" />
    <img :src="imgSrc" alt="pokemon image" class="fade-in" v-if="show" />
  </figure>
</template>

<script>
export default {
  name: "Pokemon Picture",
  props: {
    pokeId: {
      type: [Number, String],
      required: true,
      default: 151,
      validator(value) {
        return parseInt(value) > 0;
      },
    },
    show: {
      type: Boolean,
      required: true,
      default: false,
    },
  },
  computed: {
    imgSrc() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokeId}.svg`;
    },
  },
};
</script>

<style scoped>
.pokemon-picture-container {
  position: relative;
  height: 35vh;
}
.pokemon-picture-container > img {
  height: 100%;
  left: 50%;
  object-fit: contain;
  object-position: center;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}

.pokemon-picture-container > .hidden-pokemon {
  filter: brightness(0);
}
</style>
