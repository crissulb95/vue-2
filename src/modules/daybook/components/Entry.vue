<template>
  <li class="entry-item">
    <button @click="openEntry">
      <span class="date">{{ info.date }}</span>
      <h4 class="title">{{ info.title }}</h4>
      <p class="body">{{ info.body }}</p>
    </button>
  </li>
</template>

<script>
export default {
  name: 'Entry',
  props: {
    info: {
      type: Object,
      required: true,
      default: () => ({
        title: '-',
        body: '-',
        date: '-'
      })
    }
  },
  methods: {
    openEntry() {
      this.$router.push({ path: '/daybook/1' })
    }
  }
}
</script>

<style lang="scss">
.entry-item {
  flex: 0 0 calc(100% - 20px);
  margin-bottom: 10px;
  width: 100%;
  & > button {
    background-color: transparent;
    color: white;
    border: 1px solid #f7faef;
    border-radius: 10px;
    padding: 10px 20px;
    display: block;
    width: 100%;
    transition: 0.4s all ease-in-out;
    text-align: left;
    &:hover {
      background-color: white;
      color: #22577a;
    }
    & .date {
      letter-spacing: 2px;
    }
    & > .title {
      font-weight: bold;
      letter-spacing: 2px;
    }
    & > .body {
      display: -webkit-box;
      -webkit-line-clamp: 2;
      -webkit-box-orient: vertical;
      overflow: hidden;
    }
  }
}
</style>
