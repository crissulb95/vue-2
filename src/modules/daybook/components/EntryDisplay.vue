<template>
  <div class="entry-display-container">
    <router-view />
    <new-button :icon="correctIcon" />
  </div>
</template>

<script>
import { defineAsyncComponent } from '@vue/runtime-core'
export default {
  name: 'EntryDisplay',
  components: {
    NewButton: defineAsyncComponent(() =>
      import(
        /* webpackChunkName: "NewButton" */ '../../shared/components/NewButton.vue'
      )
    )
  },
  computed: {
    correctIcon() {
      if (this.$route.name === 'entry') return 'far fa-save'
      return 'fas fa-plus'
    }
  }
}
</script>

<style lang="scss">
.entry-display-container {
  border: 2px solid #f7faef;
  border-radius: 5px;
  height: 100%;
  position: relative;
}
</style>
