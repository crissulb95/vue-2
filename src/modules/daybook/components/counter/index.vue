<template>
  <div class="counter-container">
    <h3>Counter Component for VUEX</h3>
    <!-- <p><strong>Computed Count:</strong> {{ computedCount }}</p> -->
    <p><strong>mapState Count:</strong> {{ count }}</p>
    <p><strong>mapState LastMutation:</strong> {{ lastMutation }}</p>
    <p><strong>Square Count:</strong> {{ squareCount }}</p>
    <div class="btn-container">
      <button class="btn btn-primary" @click="increment">+1</button>
      <button class="btn btn-secondary" @click="incrementBy5">+5</button>
      <button class="btn btn-danger" @click="randomNumber">Random</button>
      <button
        class="btn btn-warning"
        @click="addRandomInt"
        :disabled="isLoading"
      >
        Add random
      </button>
    </div>
  </div>
</template>

<script>
import { mapActions, mapState } from 'vuex'
export default {
  name: 'Counter',
  computed: {
    // computedCount() {
    //   return this.$store.state.count
    // },
    ...mapState('pokemon', ['count', 'lastMutation', 'isLoading']),
    // ...mapState({
    //   count: (state) => state.count
    // })
    squareCount() {
      return this.$store.getters['pokemon/squareCount']
    }
  },
  methods: {
    increment() {
      this.$store.commit('pokemon/increment')
    },
    incrementBy5() {
      this.$store.commit('pokemon/incrementBy', 5)
    },
    randomNumber() {
      this.$store.commit('pokemon/randomNumber')
    },
    /* incrementRandomInt() {
      this.$store.dispatch('pokemon/incrementRandomInt')
    } */
    // ...mapActions('pokemon', ['incrementRandomInt'])
    ...mapActions('pokemon', {
      addRandomInt: 'incrementRandomInt'
    })
  }
}
</script>

<style scoped>
.counter-container {
  background-color: white;
  width: 400px;
  color: #000;
  position: absolute;
  bottom: 10px;
  right: 10px;
  box-shadow: 1px 1px 10px 2px rgba(0, 0, 0, 0.2);
  padding: 15px;
}
.counter-container h3 {
  margin-top: 0;
}

.btn-container {
  display: flex;
  width: 100%;
  flex-wrap: wrap;
  gap: 10px;
  justify-content: space-between;
}
.btn-container button {
  flex: 0 0 calc((100% / 4) - 10px);
  padding: 5px;
}
@media (max-width: 767.98px) {
  .btn-container button {
    flex: 0 0 calc((100% / 2) - 10px);
    padding: 10px;
  }
}
</style>
