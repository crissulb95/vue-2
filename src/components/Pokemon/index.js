import PokemonOptions from "./PokemonOptions.vue";
import PokemonPicture from "./PokemonPicture.vue";

export { PokemonOptions, PokemonPicture };
