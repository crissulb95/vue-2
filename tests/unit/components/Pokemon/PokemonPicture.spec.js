import { shallowMount } from "@vue/test-utils";
import { PokemonPicture } from "@/components/Pokemon";

describe("PokemonPicture Component", () => {
  /* let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(PokemonPicture);
  }); */
  test("Debería corresponder con el snapshot", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokeId: 1,
        show: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("Debería mostrar la imagen oculta del pokemon 100", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokeId: 100,
        show: true,
      },
    });
    const [img1, img2] = wrapper.findAll("img");

    expect(img1.exists()).toBeTruthy();
    expect(img2.exists()).toBe(true);
    expect(img1.classes("hidden-pokemon")).toBe(true);
    expect(img2.classes("hidden-pokemon")).toBe(false);
    expect(img1.attributes("src")).toBe(
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
    );
    console.log(img1.attributes("src"));
  });
});
