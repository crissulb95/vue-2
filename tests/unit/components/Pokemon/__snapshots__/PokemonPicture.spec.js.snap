// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture Component Deber√≠a corresponder con el snapshot 1`] = `<figure class="pokemon-picture-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon image" class="hidden-pokemon fade-in" id="hidden-pokemon"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon image" class="fade-in"></figure>`;
